pkg/graph/schema.graphql

```diff
@@ -2344,22 +2394,78 @@ type TRBAdviceLetter {
   modifiedAt: Time
 }
 
+"""
+Data needed to create a TRB admin note with the Initial Request Form category
+"""
+input TRBAdminNoteInitialRequestFormInput {
+  appliesToBasicRequestDetails: Boolean!
+  appliesToSubjectAreas: Boolean!
+  appliesToAttendees: Boolean!
+}
+
+"""
+Data needed to create a TRB admin note with the Supporting Documents category
+"""
+input TRBAdminNoteSupportingDocumentsInput {
+  documentIDs: [UUID!]!
+}
+
+"""
+Data needed to create a TRB admin note with the Advice Letter category
+"""
+input TRBAdminNoteAdviceLetterInput {
+  appliesToMeetingSummary: Boolean!
+  appliesToNextSteps: Boolean!
+  recommendationIDs: [UUID!]!
+}
+
 """
 The data needed to create a TRB admin note
+For category-specific data, rather than five separate mutations with their own input types,
+this type has nullable fields for all the categories that can have category-specific data.
+At most one of these fields should have a value.
 """
 input CreateTRBAdminNoteInput {
+  # Common fields for notes of all categories
   trbRequestId: UUID!
   category: TRBAdminNoteCategory!
   noteText: HTML!
+
+  # General Request category - no additional data
+
+  # Initial Request Form category
+  initialRequestFormData: TRBAdminNoteInitialRequestFormInput
+
+  # Supporting Documents category
+  supportingDocumentsData: TRBAdminNoteSupportingDocumentsInput
+
+  # Consult Session category - no additional data
+
+  # Advice Letter category
+  adviceLetterData: TRBAdminNoteAdviceLetterInput
```

Comment:
I think passing in nested objects for different form types introduces some issues. For anyone lacking the context, they might see this type schema and get confused on what is actually required for an admin note (without comments in FE and BE code) which undermines the benefit of the type system.

This also causes us to write validation code to ensure the correct info is being passed in per category that might not be obvious to the FE.

Reply:
Definitely will add some comments here to clarify.

Reply:
Updated in https://github.com/CMSgov/easi-app/pull/2231/commits/95bbf51eada4ee35909fd964d0c41ed5badb2013 - @mynar7, how does this look?

Reply:
Looks right so far!